File: pset6/cash.py
from cs50 import get_float

def main():
    dollars = 0

    while dollars <= 0:
        dollars = get_float("Change owed: ")

    cents = round(dollars * 100)
    counter = 0

    quarter = 25
    dime = 10
    nickel = 5
    penny = 1

    while cents > 0:
        if cents >= quarter: 
            cents -= quarter
            counter += 1
        elif cents >= dime: 
            cents -= dime
            counter += 1
        elif cents >= nickel: 
            cents -= nickel
            counter += 1
        elif cents >= penny: 
            cents -= penny
            counter += 1

    print(f'Number of coins needed for the change: {counter}')

if __name__ == "__main__":
    main()

File: pset1/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void) {
    float dollars;

    do {
        dollars = get_float("Change owed: ");
    } while (dollars < 0);

    int cents = round(dollars * 100);
    int counter = 0;

    int quarter = 25;
    int dime = 10;
    int nickel = 5;
    int penny = 1;

    while (cents > 0) {
        if (cents >= quarter) {
            cents -= quarter;
            counter++;
        } else if (cents >= dime) {
            cents -= dime;
            counter++;
        } else if (cents >= nickel) {
            cents -= nickel;
            counter++;
        } else if (cents >= penny) {
            cents -= penny;
            counter++;
        }
    }

    printf("Number of coins needed for the change: %i \n", counter);

    return 0;
}
